Automattic/wp-calypso
# New Plugin Browser styles

### Overview 
Update the header and plugin lists of the Plugins Browser screen as per oU6nEIeiKSg8CayhEtgR6b-fi-6450%3A3880

### Changes proposed in this Pull Request

* Add the themes compact and extended to Plugins Browser
* Apply the created theme to the header
* Update the search results text and show the total of results
* Add an icon on the upload button
* Proper handling of search on the mobile version 


### Differences from design
* The place of search input and search button layout on mobile 
  * It is on the right edge of the screen, to match the current behavior of the search input on mobile
* The manage and upload buttons
  * We are still awaiting a response from the design team.
* The changes placed at #57101.

### Screenshots

#### Desktop version

|Before | After|
|-------|------|
|![Screen Shot 2021-10-26 at 17 53 00](https://user-images.githubusercontent.com/5039531/138967172-00b53553-2e3c-4d4c-bc44-17bcc0fb0900.png)|![Screen Shot 2021-10-27 at 14 13 38](https://user-images.githubusercontent.com/5039531/139123210-38039bfd-c1d3-4779-8c9d-5e8cb8e994e7.png)|



#### Mobile version
|Before | After| After when searching |
|-------|------|----------------|
|![Screen Shot 2021-10-26 at 17 53 44](https://user-images.githubusercontent.com/5039531/138967408-2ba792c6-bdc5-4778-8b97-caeb5c96a216.png)|![Screen Shot 2021-10-27 at 14 14 00](https://user-images.githubusercontent.com/5039531/139123515-99fd3f75-fddd-40b1-bea1-3a867592bd55.png)|![Screen Shot 2021-10-26 at 17 54 25](https://user-images.githubusercontent.com/5039531/138967446-502beb57-3fb0-44dc-9ab8-96f0dcddb43c.png)

### Testing instructions

* Go to the plugins browse page, verify if it matches the design and screenshot above
* Search for a plugin (You can try typing 'jetpack')
* Enable the simulation of a mobile device on your browser. If you are a chrome user, you can [follow this guide](https://developer.chrome.com/docs/devtools/device-mode/)
* Go to the plugins browse page, verify if it matches the design and screenshot above
* Search for a plugin, verify if it matches the design and screenshot above


### Pending
- [x] remove the search box border on mobile when closed
- [x] keep only the upload icon in mobile
- [x] on the first-page load, the search input looks lifted up by a few pixels
- [x] fixed height for a short description aka excerpt content
- [x] the section headers have uneven margin/padding
- [x] flickering on mobile when clicking the search icon

The PR  #57101  Needs to be merged first.

---

Fixes #57103
Fixes #57067
